core/src/main/java/com/digitalpebble/storm/crawler/Metadata.java

```diff
@@ -0,0 +1,145 @@
+/**
+ * Licensed to DigitalPebble Ltd under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * DigitalPebble licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.digitalpebble.storm.crawler;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Iterator;
+import java.util.Map;
+import java.util.Map.Entry;
+
+import org.apache.commons.lang.StringUtils;
+
+/** Wrapper around Map <String,String[]> **/
+
+public class Metadata {
+
+    // TODO customize the behaviour of Kryo via annotations
+    // @BindMap(valueSerializer = IntArraySerializer.class, keySerializer =
+    // StringSerializer.class, valueClass = String[].class, keyClass =
+    // String.class, keysCanBeNull = false)
```

Comment:
yes, I think this is something we might want to look into later maybe. let's leave it as a comment for now